pallets/xcmp-handler/src/lib.rs

```diff
@@ -239,8 +239,12 @@ pub mod pallet {
 		) -> Result<(u128, u64), DispatchError> {
 			let xcm_data = XcmChainCurrencyData::<T>::get(para_id, currency_id)
 				.ok_or(Error::<T>::CurrencyChainComboNotFound)?;
+
+			let (_, target_instructions) = Self::xcm_instruction_skeleton(para_id, currency_id)?;
 			let weight = xcm_data
 				.instruction_weight
+				.checked_mul(target_instructions.len() as u64)
+				.ok_or(Error::<T>::WeightOverflow)?
```

Comment:
Is this `.ok_or` check here to throw early and skip the math of the next line?

xcm weight is tiny compared to that of transact, so if we want to add a check to the middle it’s best to apply it to `transact_encoded_call_weight` first.

Reply:
Technically, you're correct but the reality is that this will never fail under normal runtime conditions. To fail the weight would need to overflow the u64 max `18_446_744_073_709_551_615`.
The `ok_or` pattern here is a rust idiom for unwrapping the result type. `checked_mul` returns a `Result<u64, Error>` so we need to unwrap the result to get the underlying value. Also, to preserve the correct order of operations we need to do the multiplication before (or separately) from the addition.

Reply:
Gotcha, the unwrap is necessary, so let’s keep it as-is.